app/controllers/flow_controller.rb

```diff
@@ -10,18 +10,26 @@ def start
   end
 
   def show
-    @title = presenter.title
+    state = SmartAnswer::StateResolver.new(flow).state_from_response_store(response_store, node_name)
+    @presenter = FlowPresenter.new(flow, state)
+    @title = @presenter.title
 
-    if params[:node_slug] == presenter.node_slug
-      render presenter.current_node.view_template_path, formats: [:html]
+    if params[:node_slug] == @presenter.node_slug
+      render @presenter.current_node.view_template_path, formats: [:html]
     else
-      redirect_to flow_path(id: params[:id], node_slug: presenter.node_slug, params: forwarding_responses)
+      redirect_to flow_path(id: params[:id],
+                            node_slug: @presenter.node_slug,
+                            params: state.forwarding_responses)
```

Comment:
Ah, it was that way once Sean preferred it the other, now you back 😆 

I think, as forwarding_responses is a property of state, that it makes sense to use that directly rather than through presenter as it's less to look up. I don't really mind too much though - I think it remains as signs that we've still not got the cleanest object abstraction.

Reply:
We don't have the same context everywhere though. In this scenario both a state and presenter are in scope. In the places where we call forwarding_responses from @presenter a state object isn't in scope. If we go for @presenter.state.forwarding_responses then there's a fresh inconsistency with @presenter.accepted_responses.